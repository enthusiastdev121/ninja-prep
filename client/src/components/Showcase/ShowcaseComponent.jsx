import React from 'react'
import { ShowcaseDesktop } from './ShowcaseDesktop'
import { ShowcaseTablet } from './ShowcaseTablet'
import { ShowcaseMobile } from './ShowcaseMobile'
import { Desktop, Tablet, Mobile } from '../MediaQuery/DeviceSizes'

const ShowcaseComponent = () => (
    <div>
        <Desktop>
            <ShowcaseDesktop />
        </Desktop>
        <Tablet>
            <ShowcaseTablet />
        </Tablet>
        <Mobile>
            <ShowcaseMobile />
        </Mobile>
    </div>
)

export default ShowcaseComponent
export { default as ShowcaseComponent } from './ShowcaseComponent'
