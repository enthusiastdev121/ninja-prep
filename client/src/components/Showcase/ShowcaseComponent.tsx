/*
Call to action text component for customer to get involved in NinjaPrep.io
*/
import React from 'react';

import {Desktop, Tablet, Mobile} from '../../utils/DeviceSizes';
import ShowcaseDesktop from './ShowcaseDesktop';
import ShowcaseMobile from './ShowcaseMobile';
import ShowcaseTablet from './ShowcaseTablet';

const ShowcaseComponent = () => (
  <div>
    <Desktop>
      <ShowcaseDesktop />
    </Desktop>
    <Tablet>
      <ShowcaseTablet />
    </Tablet>
    <Mobile>
      <ShowcaseMobile />
    </Mobile>
  </div>
);

export default ShowcaseComponent;
