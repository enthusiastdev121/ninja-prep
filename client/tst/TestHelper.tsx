import React from 'react'
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
const history = createBrowserHistory()

interface DEVICE_WIDTHS {
    readonly DESKTOP: number
    readonly MOBILE: number
    readonly TABLET: number
}

//Widths in Pixels. Aligns with Media Query widths
export const DEVICE_TYPES: DEVICE_WIDTHS = {
    DESKTOP: 1000,
    MOBILE: 750,
    TABLET: 800
}

export const routeWrapper = (children: React.ReactChild) => {
    return <Router history={history}>{children}</Router>
}
