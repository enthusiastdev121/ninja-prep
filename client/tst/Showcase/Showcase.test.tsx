import React from 'react'
import { shallow, mount } from 'enzyme'
import { Context as ResponsiveContext } from 'react-responsive'
import { DEVICE_TYPES } from '../TestHelper'
import toJson from 'enzyme-to-json'
import ShowcaseComponent from '../../src/components/Showcase/ShowcaseComponent'

test.each(Object.entries(DEVICE_TYPES))('renders successfully on %p', (deviceType: string, deviceWidth: number) => {
    const wrappingComponent = ResponsiveContext.Provider
    const wrappingComponentProps = { value: { width: deviceWidth } }
    const mountProps = { wrappingComponent, wrappingComponentProps }

    const wrapper = shallow(<ShowcaseComponent />, mountProps)
    expect(wrapper).toBeDefined
})

test.each(Object.entries(DEVICE_TYPES))('matches snapshot on %p', (deviceType: string, deviceWidth: number) => {
    const wrappingComponent = ResponsiveContext.Provider
    const wrappingComponentProps = { value: { width: deviceWidth } }
    const mountProps = { wrappingComponent, wrappingComponentProps }

    const wrapper = mount(<ShowcaseComponent />, mountProps)
    expect(toJson(wrapper)).toMatchSnapshot()
})
