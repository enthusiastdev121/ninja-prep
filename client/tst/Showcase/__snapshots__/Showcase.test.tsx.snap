// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot on "DESKTOP" 1`] = `
<ShowcaseComponent>
  <div>
    <Desktop>
      <ShowcaseDesktop>
        <div
          className="py-5 showcase-bg"
        >
          <Container
            fluid={false}
          >
            <div
              className="container"
            >
              <div
                className="text-center pb-4"
              >
                <h3
                  className="font-weight-bold pb-2"
                >
                  Start investing in yourself now
                </h3>
                <p
                  className="lead"
                >
                  Now sure where to begin? Click on a problem and starting learning!
                </p>
              </div>
            </div>
          </Container>
        </div>
      </ShowcaseDesktop>
    </Desktop>
    <Tablet />
    <Mobile />
  </div>
</ShowcaseComponent>
`;

exports[`matches snapshot on "MOBILE" 1`] = `
<ShowcaseComponent>
  <div>
    <Desktop />
    <Tablet />
    <Mobile>
      <ShowcaseMobile>
        <div
          className="py-5 showcase-bg"
        >
          <Container
            fluid={false}
          >
            <div
              className="container"
            >
              <div
                className="text-center pb-4"
              >
                <h3
                  className="font-weight-bold pb-2"
                >
                  Start investing in yourself now
                </h3>
                <p
                  className="lead"
                >
                  Now sure where to begin? Click on a problem and starting learning!
                </p>
              </div>
            </div>
          </Container>
        </div>
      </ShowcaseMobile>
    </Mobile>
  </div>
</ShowcaseComponent>
`;

exports[`matches snapshot on "TABLET" 1`] = `
<ShowcaseComponent>
  <div>
    <Desktop />
    <Tablet>
      <ShowcaseTablet>
        <div
          className="py-5 showcase-bg"
        >
          <Container
            fluid={false}
          >
            <div
              className="container"
            >
              <div
                className="text-center pb-4"
              >
                <h3
                  className="font-weight-bold pb-2"
                >
                  Start investing in yourself now
                </h3>
                <p
                  className="lead"
                >
                  Now sure where to begin? Click on a problem and starting learning!
                </p>
              </div>
            </div>
          </Container>
        </div>
      </ShowcaseTablet>
    </Tablet>
    <Mobile />
  </div>
</ShowcaseComponent>
`;
